### PR TITLE
added hmac support

### PR DESCRIPTION
**Pull request by c9dong**

### CODE REVIEW

File: api/shared.js [hunk 1]
```diff
@@ -1,39 +1,41 @@
 var router = require('koa-router')();
 var bodyParser = require('koa-body')();
-var fetch = require('node-fetch');
+// var fetch = require('node-fetch');
```
Comment (jasonf7) [inline api/shared.js]: pls remove

File: authFetch.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  return fetch(url, data);
+}
+
+function addAuthToken(data) {
+  if (!data) {
+    data = {};
+  }
+  if (!data.headers) {
+    data.headers = {};
+  }
```
Comment (jasonf7) [inline authFetch.js:16]: Why not
`
if (!data || !data.headers) {
    data = {
        headers: {}
    };
}
`

**Review by jasonf7 — CHANGES_REQUESTED** [review]
m

File: authFetch.js (hunk 2 above)
Comment (c9dong) [reply authFetch.js:16]: data can contain other information, so I can't overwrite them

**Review by jasonf7 — APPROVED** [review]

**Pull request merged**